### PR TITLE
Implemented UI element for healthy control

### LINKED ISSUE #2: Expose query variables with UI elements
The UI should expose the same variables with UI elements (sliders, dropdowns, etc) that the API is exposing ([see here](https://github.com/neurobagel/bagelapi/issues/31)).

- [x] #14
- [x] #15
- [x] #16
- [x] #17
- [x] #18
- [x] #19
- [x] #20
  - e.g. whether a participant is a control subject or not (as implemented in the API)

Each of these should:
- Implement the UI element as a vue component
- Display the component in the QueryFrom.vue component
- Connect the element to an (asynchronous) API request
- Test the implementation with a component test
- have a short line added to the README about what the UI element queries (like an early draft of user docs)
  - e.g.: `age: select the minimal and/or maximal age (in years) of a participant that should still be included in the results`
- provide options for sex, diagnosis, imaging modality, assessment tools

### LINKED ISSUE #20: Implement UI for healthy control
- [x] Implement using `slots` in `CategoricalField.vue`
- [x] Implement `slot` for the diagnosis field in `QueryForm.vue`
- [x] Integrate healthy control checkbox in `QueryForm.vue`
- [x] Update `submitQuery` method in `QueryForm.vue`
- [x] Update `QueryForm.cy.js
- [x] Implement test for disabling diagnosis field when healthy control checkbox is checked in QueryForm.cy.js
- [x] Refactor `CategoricalField.cy.js
- [x] Implement test for displaying `control` slot in `CategoricalField.cy.js`
- [x] Have a short line added to the README about what the UI element queries

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #20 
Closes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,11 +75,12 @@ You can verify the tool is running once you receive info messages from Nuxt rega
 
 ## Usage
 
-To define a cohort, set your inclusion criteria using the following query fields:
+To define a cohort, set your inclusion criteria using the following:
```
Comment (surchs) [inline README.md:78]: Now it reads like the last word is missing.

File: components/QueryForm.vue [hunk 2]
```diff
@@ -37,8 +37,21 @@
           data-cy="diagnosis-field"
           :default-selected="Object.keys(categoricalOptions.Diagnosis)[0]"
           :options="Object.keys(categoricalOptions.Diagnosis)"
+          :disabled="is_control"
           @update-categorical-field="updateField"
-        />
+        >
+          <template #control>
```
Comment (surchs) [inline components/QueryForm.vue:43]: Very nice! Now that you're an expert on scoped slots, you should explain to the rest of the team in 5 minutes how they work!

File: components/QueryForm.vue [hunk 3]
```diff
@@ -37,8 +37,21 @@
...
+            <b-form-group class="col-md-7">
+              <input
```
Comment (surchs) [inline components/QueryForm.vue:45]: You may have explained this before, but is there a reason we cannot go with https://bootstrap-vue.org/docs/components/form-checkbox and get the whole thing in one? Not a strong preference, mostly thinking about consistency.

File: cypress/component/CategoricalField.cy.js [hunk 4]
```diff
@@ -30,13 +25,18 @@ describe('Categorical field', () => {
       listeners: {
         'update-categorical-field': cy.spy().as('spy'),
       },
-      propsData: {
-        name: fields.sex.name,
-        defaultSelected: fields.sex.defaultSelected,
-        options: fields.sex.options,
+      propsData: props,
+    });
+    cy.get('[data-cy="Diagnosis-select"]').type(props.options[2]).type('{enter}');
```
Comment (surchs) [inline cypress/component/CategoricalField.cy.js:30]: very nice!

File: cypress/component/CategoricalField.cy.js [hunk 5]
```diff
@@ -30,13 +25,18 @@ describe('Categorical field', () => {
...
+    cy.get('@spy').should('have.been.calledWith', props.name, props.options[2]);
+  });
+  it('Displays control slot', () => {
+    cy.mount(CategoricalField, {
+      propsData: props,
+      slots: {
```
Comment (surchs) [inline cypress/component/CategoricalField.cy.js:36]: oh wow, I was unaware of this possibility. Yeah, you definitely have to give us a quick summary of what you've learned about scoped slots!

**Review by surchs — APPROVED** [review]
Very cool work @rmanaem! This looks great - and closes the last feature for the milestone! Calls for celebration :tada: !

I left a (very minor) comment, take a look. But good to go :tada: !

File: README.md (hunk 1 above)
Comment (rmanaem) [reply README.md:78]: How about `using the following elements`?
I removed the field cause a checkbox is not a field 😄

File: components/QueryForm.vue (hunk 3 above)
Comment (rmanaem) [reply components/QueryForm.vue:45]: I tried using that first but it throws the styles off.
As for other places like `select all` checkbox, I think the component is made to work in a form and doesn't behave the way a standalone `input` does.

File: README.md (hunk 1 above)
Comment (surchs) [reply README.md:78]: ah, I see. Yeah, that makes sense. I wasn't sure if the edit was intentional. I think it also works as is, but reads a bit technical

File: components/QueryForm.vue (hunk 2 above)
Comment (rmanaem) [reply components/QueryForm.vue:43]: Well that's not a scoped slot but sure I'd be happy to 😄

**Pull request merged**